bin/ch/ch.cpp

```diff
@@ -196,97 +196,77 @@ HANDLE GetFileHandle(LPCWSTR filename)
     return GetStdHandle(STD_OUTPUT_HANDLE);
 }
 
-HRESULT CreateLibraryByteCodeHeader(LPCSTR contentsRaw, JsFinalizeCallback contentsRawFinalizeCallback, DWORD lengthBytes, LPCWSTR bcFullPath, LPCSTR libraryNameNarrow)
+HRESULT CreateLibraryByteCode(const char* contentsRaw)
 {
-    HANDLE bcFileHandle = nullptr;
     JsValueRef bufferVal;
     BYTE *bcBuffer = nullptr;
     unsigned int bcBufferSize = 0;
-    DWORD written;
-    // For validating the header file against the library file
-    auto outputStr =
-        "//-------------------------------------------------------------------------------------------------------\n"
-        "// Copyright (C) Microsoft. All rights reserved.\n"
-        "// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.\n"
-        "//-------------------------------------------------------------------------------------------------------\n"
-        "#if 0\n";
-
-    std::string normalizedContentStr;
-    char* nextToken = nullptr;
-    char* token = strtok_s((char*)contentsRaw, "\r", &nextToken);
-    while (token)
-    {
-        normalizedContentStr.append(token);
-        token = strtok_s(nullptr, "\r", &nextToken);
-    }
-    // We no longer need contentsRaw, so call the finalizer for it if one was provided
-    if (contentsRawFinalizeCallback != nullptr)
-    {
-        contentsRawFinalizeCallback((void*)contentsRaw);
-    }
-
-    const char* normalizedContent = normalizedContentStr.c_str();
-    // We still need contentsRaw after this, so pass a null finalizeCallback into it
-    HRESULT hr = GetSerializedBuffer(normalizedContent, nullptr, &bufferVal);
-
-    IfFailedGoLabel((hr), ErrorRunFinalize);
-
-    IfJsrtErrorHRLabel(ChakraRTInterface::JsGetArrayBufferStorage(bufferVal, &bcBuffer, &bcBufferSize), ErrorRunFinalize);
-
-    bcFileHandle = GetFileHandle(bcFullPath);
-    IfFalseGo(bcFileHandle != INVALID_HANDLE_VALUE && bcFileHandle != nullptr);
+    HRESULT hr = E_FAIL;
+    
+    // Windows can't do the below with printf - so use windows API on windows but printf on posix
+    // Also on windows loading the file may have introduced CR line endings so adjust
+    #ifdef _WIN32
+        std::string normalisedContentStr;
+        char* nextToken = nullptr;
+        char* token = strtok_s((char*)contentsRaw, "\r", &nextToken);
+        while (token)
+        {
+            normalisedContentStr.append(token);
+            token = strtok_s(nullptr, "\r", &nextToken);
+        }
+        const char* normalisedContent = normalisedContentStr.c_str();
```

Comment:
Remove CR line endings from the input js - the input files do not contain any BUT within azure when they're loaded on windows they are inserted - not sure why as it doesn't happen offline on macOS or windows 10.

Reply:
Ugh, this sounds like `core.autocrlf` is being set on the CI machines. The `.gitattributes` file in this repo badly needs to be expanded.

Reply:
Thanks for the tip - I've updated .gitattributes to include js files within lib\ and was able to drop this line ending normalising code.